pyteal/compiler/compiler.py

```diff
@@ -237,6 +239,7 @@ def compileTeal(
     version: int = DEFAULT_PROGRAM_VERSION,
     assembleConstants: bool = False,
     optimize: OptimizeOptions = None,
+    debug: bool = False,
```

Comment:
Debug comments look just like a source map to me. I think it would be better to actually return a distinct source map object at this point.